components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -155,6 +166,33 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
         }
     }
 
+    protected void associateUser(String username, String userStoreDomain, String tenantDomain, String subject,
+                                 String idp) throws FrameworkException {
+
+        String usernameWithUserstoreDomain = UserCoreUtil.addDomainToName(username, userStoreDomain);
+        try {
+            // start tenant flow
+            FrameworkUtils.startTenantFlow(tenantDomain);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(usernameWithUserstoreDomain);
+
+            if (!StringUtils.isEmpty(idp) && !StringUtils.isEmpty(subject)) {
```

Comment:
What if IDP subject is not available ? It will just go through this silently. Probably this may not be a case we have to consider since by this point probably the IDP and subject shouldn't be empty. Still shall we add a debug log for the sake of safety.